backend/dataall/modules/dataset_sharing/services/share_object_service.py

```diff
@@ -65,13 +100,26 @@ def create_share_object(
                     environment.environmentUri
                 )
                 principal_iam_role_name = consumption_role.IAMRoleName
+                ShareObjectService.check_if_target_role_has_policies_attached(
+                    consumption_role.IAMRoleName,
```

Comment:
minor: pass `principal_iam_role_name` here instead of `consumption_role.IAMRoleName`

Reply:
changed